src/test/plugins/memory_tracking_plugin_system_test.cpp

```diff
@@ -0,0 +1,98 @@
+#include "base_test.hpp"
+#include "hyrise.hpp"
+#include "lib/utils/plugin_test_utils.hpp"
+#include "utils/memory_resource_manager.hpp"
+#include "../../plugins/memory_tracking_plugin.hpp"
+
+namespace opossum {
+
+class MemoryTrackingPluginSystemTest : public BaseTest {
+ protected:
+  MemoryResourceManager* _memory_resource_manager;
+  void SetUp() override {
+    _memory_resource_manager = &Hyrise::get().memory_resource_manager;
+
+    // Load the memory tracking plugin.
+    auto& pm = Hyrise::get().plugin_manager;
+    pm.load_plugin(build_dylib_path("libhyriseMemoryTrackingPlugin"));
+  }
+  void TearDown() override {
```

Comment:
`Hyrise::reset()` is already called by the destructor of `BaseTest`. You don't have to call it in `TearDown()`.